Fix missing sort price and/or product limit/offset when ordering by price on listing page

soulardromain opened the pull request:
<!--
Thank you for contributing to the PrestaShop project! 

Please take the time to edit the "Answers" rows with the necessary information.

Check out our contribution guidelines to find out how to complete it:
https://devdocs.prestashop.com/1.7/contribute/contribution-guidelines/#pull-requests
 -->

| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | Search page: fix sort price using Tools::orderbyPrice function and array_slice for pagination. Category page, manufacturer page, supplier page, new products page, prices drop page: fix sort price using array_slice for pagination. I based on the best sales page.
| Type?         | bug fix
| Category?     | FO
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? |  #17212
| How to test?  | Add a specific price on a product and sort any listing page except best sales page and use pagination.

<!-- Click the form's "Preview" button to make sure the table is functional in GitHub. Thank you! -->

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/prestashop/prestashop/16698)
<!-- Reviewable:end -->

rsoulard-prolaser committed:
Fix missing sort price and/or product limit/offset when ordering by price on listing page

rsoulard-prolaser committed:
pretty fix

matks committed:
Use _PS_PRICE_COMPUTE_PRECISION_

PierreRambaud commented on classes/Category.php:
```diff
@@ -1049,19 +1051,20 @@ public function getProducts(
 
         if (!$result) {
-            return array();
+            return false;
```
Why changing this?

soulardromain replied:
I don't remember, it was probably a mistake when I copied code from the bestsellers page, do I have to commit to cancel this line?

PierreRambaud replied:
Yes, since there is no reason to commit this line :)

soulardromain replied:
ok done

PierreRambaud commented on classes/Category.php:
```diff
@@ -1049,19 +1051,20 @@ public function getProducts(
             $sql .= ' ORDER BY RAND() LIMIT ' . (int) $randomNumberProducts;
-        } else {
-            $sql .= ' ORDER BY ' . (!empty($orderByPrefix) ? $orderByPrefix . '.' : '') . '`' . bqSQL($orderyBy) . '` ' . pSQL($orderWay) . '
+        } elseif ($finalOrderBy != 'orderprice') {
```
So there is no `ORDER BY`  when ordering by price?

soulardromain replied:
Ordering by price is at line 1065

PierreRambaud commented:
Hi and thanks for your contribution. Can you create an issue to be sure we can reproduce the problem? :)
Kind regards

soulardromain commented:
Hi, i have create an issue : #17212

sowbiba committed:
fix: closes #17345 - Get real attribute combination quantity to disable add to cart button

sowbiba committed:
fix: closes #17437 - Refresh product attributes

nesrineabdmouleh committed:
Choose quantity discounts based on

Commit:
Css translations.css and product_page.css do not exist

Progi1984 committed:
Fixed display for first cell (select all & input) in grid

atomiix committed:
Add missing deprecation informations

(cherry picked from commit 1a75f85a3468b22969799a1b91e4a7ca73d420fb)

nesrineabdmouleh committed:
Add some fixes

atomiix committed:
Fix invoice address changed when changing shipping address

atomiix committed:
Rebuild assets

nesrineabdmouleh committed:
Fix add product to cart function

atomiix committed:
Change useless email check

(cherry picked from commit 78d0dfc4f3d56a047084e139dc53ff69e66f7368)

nesrineabdmouleh committed:
Fix selector name

Commit:
Closing brackets causin a safari bug

Commit:
Lint problem

Commit:
csfixer

NeOMakinG commented:
Hi @soulardromain, there are some conflicts on the PR :)

nesrineabdmouleh committed:
Add some fixes

soulardromain commented:
Hi @NeOMakinG, conflicts are corrected

NeOMakinG commented on classes/Product.php:
```diff
@@ -2931,18 +2938,23 @@ public static function getPricesDrop(
         }
 
-        if ($order_by == 'price') {
+        if ($finalOrderBy == 'price') {
```
```suggestion
        if ($finalOrderBy === 'price') {
```

NeOMakinG commented on classes/Search.php:
```diff
@@ -426,11 +429,20 @@ public static function find(
-				' . ($order_by ? 'ORDER BY  ' . $alias . $order_by : '') . ($order_way ? ' ' . $order_way : '') . '
+				GROUP BY product_shop.id_product';
+
+        if ($finalOrderBy != 'price') {
```
```suggestion
        if ($finalOrderBy !== 'price') {
```

NeOMakinG commented on classes/Supplier.php:
```diff
@@ -391,18 +393,23 @@ public static function getProducts(
         }
 
-        if ($orderBy == 'price') {
+        if ($finalOrderBy == 'price') {
```
```suggestion
        if ($finalOrderBy === 'price') {
```

NeOMakinG commented on classes/Supplier.php:
```diff
@@ -391,18 +393,23 @@ public static function getProducts(
-				ORDER BY ' . $alias . pSQL($orderBy) . ' ' . pSQL($orderWay) . '
+				GROUP BY ps.id_product';
+
+        if ($finalOrderBy != 'price') {
```
```suggestion
        if ($finalOrderBy !== 'price') {
```

NeOMakinG commented on classes/Search.php:
```diff
@@ -426,11 +429,20 @@ public static function find(
+
         $result = $db->executeS($sql, true, false);
 
+        if ($finalOrderBy == 'price') {
```
```suggestion
        if ($finalOrderBy === 'price') {
```

NeOMakinG commented on classes/Category.php:
```diff
@@ -991,30 +991,32 @@ public function getProducts(
 
-        if ($orderyBy == 'price') {
-            $orderyBy = 'orderprice';
+        if ($orderBy == 'price') {
```
```suggestion
        if ($orderBy === 'price') {
```

NeOMakinG commented on classes/Category.php:
```diff
@@ -991,30 +991,32 @@ public function getProducts(
-            $orderyBy = 'name';
-        } elseif ($orderyBy == 'position') {
+            $orderBy = 'name';
+        } elseif ($orderBy == 'position') {
```
```suggestion
        } elseif ($orderBy === 'position') {
```

NeOMakinG commented on classes/Category.php:
```diff
@@ -991,30 +991,32 @@ public function getProducts(
+        } elseif ($orderBy == 'name') {
             $orderByPrefix = 'pl';
-        } elseif ($orderyBy == 'manufacturer' || $orderyBy == 'manufacturer_name') {
+        } elseif ($orderBy == 'manufacturer' || $orderBy == 'manufacturer_name') {
```
```suggestion
        } elseif ($orderBy === 'manufacturer' || $orderBy === 'manufacturer_name') {
```

NeOMakinG commented on classes/Category.php:
```diff
@@ -991,30 +991,32 @@ public function getProducts(
+        if ($orderBy == 'id_product' || $orderBy == 'date_add' || $orderBy == 'date_upd') {
             $orderByPrefix = 'p';
-        } elseif ($orderyBy == 'name') {
+        } elseif ($orderBy == 'name') {
```
```suggestion
        } elseif ($orderBy === 'name') {
```

NeOMakinG commented on classes/Category.php:
```diff
@@ -991,30 +991,32 @@ public function getProducts(
 
         $orderByPrefix = false;
-        if ($orderyBy == 'id_product' || $orderyBy == 'date_add' || $orderyBy == 'date_upd') {
+        if ($orderBy == 'id_product' || $orderBy == 'date_add' || $orderBy == 'date_upd') {
```
```suggestion
        if ($orderBy === 'id_product' || $orderBy === 'date_add' || $orderBy === 'date_upd') {
```

NeOMakinG commented on classes/Category.php:
```diff
@@ -1049,8 +1051,8 @@ public function getProducts(
             $sql .= ' ORDER BY RAND() LIMIT ' . (int) $randomNumberProducts;
-        } else {
-            $sql .= ' ORDER BY ' . (!empty($orderByPrefix) ? $orderByPrefix . '.' : '') . '`' . bqSQL($orderyBy) . '` ' . pSQL($orderWay) . '
+        } elseif ($finalOrderBy != 'orderprice') {
```
```suggestion
        } elseif ($finalOrderBy !== 'orderprice') {
```

NeOMakinG commented on classes/Category.php:
```diff
@@ -1060,8 +1062,9 @@ public function getProducts(
         }
 
-        if ($orderyBy == 'orderprice') {
+        if ($finalOrderBy == 'orderprice') {
```
```suggestion
        if ($finalOrderBy === 'orderprice') {
```

NeOMakinG commented on classes/Manufacturer.php:
```diff
@@ -484,18 +486,23 @@ public static function getProducts(
-				GROUP BY p.id_product
+				GROUP BY p.id_product';
+
+        if ($finalOrderBy != 'price') {
```
```suggestion
        if ($finalOrderBy !== 'price') {
```

NeOMakinG commented on classes/Manufacturer.php:
```diff
@@ -484,18 +486,23 @@ public static function getProducts(
         }
 
-        if ($orderBy == 'price') {
+        if ($finalOrderBy == 'price') {
```
```suggestion
        if ($finalOrderBy === 'price') {
```

NeOMakinG commented on classes/Product.php:
```diff
@@ -2670,8 +2672,10 @@ public static function getNewProducts($id_lang, $page_number = 0, $nb_products =
 
-        $sql->orderBy((isset($order_by_prefix) ? pSQL($order_by_prefix) . '.' : '') . '`' . pSQL($order_by) . '` ' . pSQL($order_way));
-        $sql->limit($nb_products, (int) (($page_number - 1) * $nb_products));
+        if ($finalOrderBy != 'price') {
```
```suggestion
        if ($finalOrderBy !== 'price') {
```

NeOMakinG commented on classes/Product.php:
```diff
@@ -2685,8 +2689,9 @@ public static function getNewProducts($id_lang, $page_number = 0, $nb_products =
         }
 
-        if ($order_by == 'price') {
+        if ($finalOrderBy == 'price') {
```
```suggestion
        if ($finalOrderBy === 'price') {
```

NeOMakinG reviewed (changes requested):
Please prefer using strict operators :)

Thanks for your contribution

nesrineabdmouleh committed:
Test display remaining quantity

soulardromain committed:
Update classes/Category.php

Co-Authored-By: SZCZUPAK Valentin <valentin.szczupak@prestashop.com>

soulardromain committed:
Update classes/Category.php

Co-Authored-By: SZCZUPAK Valentin <valentin.szczupak@prestashop.com>

soulardromain committed:
Update classes/Category.php

Co-Authored-By: SZCZUPAK Valentin <valentin.szczupak@prestashop.com>

soulardromain committed:
Update classes/Category.php

Co-Authored-By: SZCZUPAK Valentin <valentin.szczupak@prestashop.com>

soulardromain committed:
Update classes/Product.php

Co-Authored-By: SZCZUPAK Valentin <valentin.szczupak@prestashop.com>

soulardromain committed:
Update classes/Search.php

Co-Authored-By: SZCZUPAK Valentin <valentin.szczupak@prestashop.com>

soulardromain committed:
Update classes/Supplier.php

Co-Authored-By: SZCZUPAK Valentin <valentin.szczupak@prestashop.com>

soulardromain committed:
Update classes/Supplier.php

Co-Authored-By: SZCZUPAK Valentin <valentin.szczupak@prestashop.com>

mvorisek committed:
Add support for non-unity MySQL autoincrement

soulardromain committed:
Update classes/Search.php

Co-Authored-By: SZCZUPAK Valentin <valentin.szczupak@prestashop.com>

soulardromain committed:
Update classes/Category.php

Co-Authored-By: SZCZUPAK Valentin <valentin.szczupak@prestashop.com>

soulardromain committed:
Update classes/Manufacturer.php

Co-Authored-By: SZCZUPAK Valentin <valentin.szczupak@prestashop.com>

soulardromain committed:
Update classes/Manufacturer.php

Co-Authored-By: SZCZUPAK Valentin <valentin.szczupak@prestashop.com>

soulardromain committed:
Update classes/Product.php

Co-Authored-By: SZCZUPAK Valentin <valentin.szczupak@prestashop.com>

soulardromain committed:
Update classes/Product.php

Co-Authored-By: SZCZUPAK Valentin <valentin.szczupak@prestashop.com>

soulardromain committed:
Update classes/Category.php

Co-Authored-By: SZCZUPAK Valentin <valentin.szczupak@prestashop.com>

soulardromain committed:
Update classes/Category.php

Co-Authored-By: SZCZUPAK Valentin <valentin.szczupak@prestashop.com>

soulardromain committed:
Update classes/Category.php

Co-Authored-By: SZCZUPAK Valentin <valentin.szczupak@prestashop.com>

mvorisek committed:
Fix typo - no function change

mvorisek committed:
Fix & remove unneeded reference operators

tdavidsonas88 committed:
moving forward with UI design

nesrineabdmouleh committed:
Add some fixes

tdavidsonas88 committed:
use existing Order functions to get prev/next Order

zuk3975 committed:
fix strange docblock

boubkerbribri committed:
update base context

Co-Authored-By: Nesrine Abdmouleh <nesrine-abdmouleh.ext@prestashop.com>

Commit:
delete default value for sort direction

Commit:
Merge branch 'SortCustomersTest' of github.com:boubkerbribri/PrestaShop into SortCustomersTest

jolelievre committed:
Fix cancel product type translation domain

tdavidsonas88 committed:
OrderLinkPresenter moved to the Adapter/Presenter/Order folder

tdavidsonas88 committed:
btns style changed according to other btns

Commit:
Merge pull request #17757 from jolelievre/fix-trads

Fix cancel product type translation domain

rokaszygmantas committed:
Merge pull request #17305 from zuk3975/fix-supplier-add

Adds optional DNI field and missing post_code handling in supplier form

tdavidsonas88 committed:
alignment of left/right btns

nesrineabdmouleh committed:
Fix const name

Progi1984 committed:
Merge pull request #17700 from atomiix/fix-address-change

Wrong address update when changing the shipping adresse of an order

tdavidsonas88 committed:
alignment right of the block

tdavidsonas88 committed:
bootstrap blocks alignment

zuk3975 committed:
move css from backend

matks committed:
Merge pull request #17733 from matks/fix-sql-update

Unwanted spaces in 1.7.7.0.sql file

Commit:
Revert "csfixer"

This reverts commit 2458553a60958f9ac531cbf4c524ae5a8d043246.

atomiix committed:
Merge pull request #17116 from mvorisek/patch-1

Unnecessary localhost string when using getRemoteAddr

atomiix committed:
Merge pull request #17535 from atomiix/deleted-customer-order

Fix deleted customer's order preview/details

Commit:
Add demo data

Commit:
Add test import localization pack

Commit:
Fix updateCheckboxValue function

jolelievre committed:
Update catalog for 177 version

Commit:
fix comments

Progi1984 committed:
Merge pull request #17577 from PierreRambaud/fix/email-enumeration

Prevent email enumeration

tdavidsonas88 committed:
disabled styles changed of the button

Commit:
Fix selector name

Progi1984 committed:
Merge pull request #17568 from atomiix/order-price-font-size

Order price font size

Progi1984 committed:
Merge pull request #17497 from jf-viguier/increase-bo-cookielifetime2

Increase bo cookie lifetime from 15 to 60 minutes

tdavidsonas88 committed:
changes after code review

tdavidsonas88 committed:
changes after code review

tdavidsonas88 committed:
changes after code review

tdavidsonas88 committed:
changes after code review

tdavidsonas88 committed:
fixes on naming and types

atomiix committed:
Merge pull request #17645 from atomiix/backport-16927

Backport #16927

zuk3975 committed:
remove irrelevant spaces from template

tdavidsonas88 committed:
removed the not needed legacy Order object form the OrderLinkPresenter.php

Progi1984 committed:
Merge pull request #17528 from tdavidsonas88/1.7.7.x-bo-routes-controllers-follow-naming-conventions

Fix all 1.7.7.x controllers and routes to comply with conventions

Progi1984 committed:
Merge pull request #17702 from matthieu-rolland/make-mail-failure-nonblocking

Order Page: Keep email failure from blocking product cancellation (and some refactor)

Commit:
Add test sort addresses

tdavidsonas88 committed:
refactored to simplify the code - OrderLinkPresenter.php not needed

tdavidsonas88 committed:
explanatory comments added

SimonGrn committed:
Merge pull request #17741 from boubkerbribri/SortContactsTest

Functional tests - Add test 'Sort contacts'

SimonGrn committed:
Merge pull request #17756 from boubkerbribri/SortCustomersTest

Functional tests - Add test 'Sort customers'

Commit:
Update comments

Progi1984 commented on classes/Category.php:
```diff
@@ -991,30 +991,32 @@ public function getProducts(
             $nbDaysNewProduct = 20;
         }
 
+        $finalOrderBy = $orderBy;
```
```suggestion
```

Progi1984 commented on classes/Category.php:
```diff
@@ -1049,8 +1051,8 @@ public function getProducts(
             $sql .= ' ORDER BY RAND() LIMIT ' . (int) $randomNumberProducts;
-        } else {
-            $sql .= ' ORDER BY ' . (!empty($orderByPrefix) ? $orderByPrefix . '.' : '') . '`' . bqSQL($orderyBy) . '` ' . pSQL($orderWay) . '
+        } elseif ($finalOrderBy !== 'orderprice') {
```
```suggestion
        } elseif ($orderBy !== 'orderprice') {
```

Progi1984 commented on classes/Category.php:
```diff
@@ -1060,8 +1062,9 @@ public function getProducts(
         }
 
-        if ($orderyBy == 'orderprice') {
+        if ($finalOrderBy === 'orderprice') {
```
```suggestion
        if ($orderBy === 'orderprice') {
```

Progi1984 commented on classes/Manufacturer.php:
```diff
@@ -445,6 +445,8 @@ public static function getProducts(
         }
 
+        $finalOrderBy = $orderBy;
+
```
```suggestion
```

Progi1984 commented on classes/Manufacturer.php:
```diff
@@ -484,18 +486,23 @@ public static function getProducts(
-				GROUP BY p.id_product
+				GROUP BY p.id_product';
+
+        if ($finalOrderBy !== 'price') {
```
```suggestion
        if ($orderBy !== 'price') {
```

Progi1984 commented on classes/Manufacturer.php:
```diff
@@ -484,18 +486,23 @@ public static function getProducts(
         }
 
-        if ($orderBy == 'price') {
+        if ($finalOrderBy === 'price') {
```
```suggestion
        if ($orderBy === 'price') {
```

Progi1984 commented on classes/Product.php:
```diff
@@ -2620,6 +2620,8 @@ public static function getNewProducts($id_lang, $page_number = 0, $nb_products =
         }
 
+        $finalOrderBy = $order_by;
+
```
```suggestion
```

Progi1984 commented on classes/Product.php:
```diff
@@ -2670,8 +2672,10 @@ public static function getNewProducts($id_lang, $page_number = 0, $nb_products =
 
-        $sql->orderBy((isset($order_by_prefix) ? pSQL($order_by_prefix) . '.' : '') . '`' . pSQL($order_by) . '` ' . pSQL($order_way));
-        $sql->limit($nb_products, (int) (($page_number - 1) * $nb_products));
+        if ($finalOrderBy !== 'price') {
```
```suggestion
        if ($order_by !== 'price') {
```

Progi1984 commented on classes/Product.php:
```diff
@@ -2685,8 +2689,9 @@ public static function getNewProducts($id_lang, $page_number = 0, $nb_products =
         }
 
-        if ($order_by == 'price') {
+        if ($finalOrderBy === 'price') {
```
```suggestion
        if ($order_by === 'price') {
```

Progi1984 commented on classes/Product.php:
```diff
@@ -2901,6 +2906,8 @@ public static function getPricesDrop(
         }
 
+        $finalOrderBy = $order_by;
+
```
```suggestion
```

Progi1984 commented on classes/Product.php:
```diff
@@ -2931,18 +2938,23 @@ public static function getPricesDrop(
-        LIMIT ' . (int) (($page_number - 1) * $nb_products) . ', ' . (int) $nb_products;
+        ' . $sql_groups;
+
+        if ($finalOrderBy != 'price') {
```
```suggestion
        if ($order_by != 'price') {
```

Progi1984 commented on classes/Product.php:
```diff
@@ -2931,18 +2938,23 @@ public static function getPricesDrop(
         }
 
-        if ($order_by == 'price') {
+        if ($finalOrderBy === 'price') {
```
```suggestion
        if ($order_by === 'price') {
```

Progi1984 commented on classes/Search.php:
```diff
@@ -401,6 +401,9 @@ public static function find(
         }
+
+        $finalOrderBy = $order_by;
+
```
```suggestion
```

Progi1984 commented on classes/Search.php:
```diff
@@ -426,11 +429,20 @@ public static function find(
-				' . ($order_by ? 'ORDER BY  ' . $alias . $order_by : '') . ($order_way ? ' ' . $order_way : '') . '
+				GROUP BY product_shop.id_product';
+
+        if ($finalOrderBy !== 'price') {
```
```suggestion
        if ($order_by !== 'price') {
```

Progi1984 commented on classes/Search.php:
```diff
@@ -426,11 +429,20 @@ public static function find(
+
         $result = $db->executeS($sql, true, false);
 
+        if ($finalOrderBy === 'price') {
```
```suggestion
        if ($order_by === 'price') {
```

Progi1984 commented on classes/Supplier.php:
```diff
@@ -347,6 +347,8 @@ public static function getProducts(
         }
 
+        $finalOrderBy = $orderBy;
+
```
```suggestion
```

Progi1984 commented on classes/Supplier.php:
```diff
@@ -391,18 +393,23 @@ public static function getProducts(
-				ORDER BY ' . $alias . pSQL($orderBy) . ' ' . pSQL($orderWay) . '
+				GROUP BY ps.id_product';
+
+        if ($finalOrderBy !== 'price') {
```
```suggestion
        if ($orderBy !== 'price') {
```

Progi1984 commented on classes/Supplier.php:
```diff
@@ -391,18 +393,23 @@ public static function getProducts(
         }
 
-        if ($orderBy == 'price') {
+        if ($finalOrderBy === 'price') {
```
```suggestion
        if ($orderBy === 'price') {
```

tdavidsonas88 committed:
explanatory comments added in an appropriate place, pretty-CI failing

nesrineabdmouleh committed:
Test send welcome email

nesrineabdmouleh committed:
Fix Email class name

tdavidsonas88 committed:
Category update position test refactored to use generated position

PierreRambaud committed:
Merge pull request #17738 from davidglezz/patch-9

Simplify mathematical operation in StockAvailable::setQuantity()

tdavidsonas88 committed:
to rerun pretty-ci

Commit:
Add sort and position functions

Commit:
Add categories tests

davidglezz committed:
Fix X-Forwarded-For not supported properly

Range must be 172.16/12 instead 172.16/16 (https://tools.ietf.org/html/rfc1918#section-3)

mvorisek committed:
Merge branch 'behat-fix-failures-with-different-auto-increment' into gh_pr_fix_autoincrement

nesrineabdmouleh committed:
Configure currency restriction

Add some fixes

nesrineabdmouleh committed:
Delete comment

Commit:
Fix new wording without domain

Commit:
Move wording from messages to Admin.Catalog.Feature as per #17796

Commit:
Merge pull request #17763 from jolelievre/update-catalog-177

Update catalog for 1.7.7.0

matthieu-rolland committed:
Merge pull request #17796 from eternoendless/fix-wording-domain

Fix new wording without domain

nesrineabdmouleh committed:
Add some fixes

matthieu-rolland committed:
always display button after input

matthieu-rolland committed:
asset build

matthieu-rolland committed:
Merge pull request #17752 from tdavidsonas88/17648-prev-next-order-button

Add Prev/Next order button in the Order page

boubkerbribri committed:
update step description

Co-Authored-By: GoT <PierreRambaud@users.noreply.github.com>

matthieu-rolland committed:
only update order carrier if it exists

Commit:
fix step description

SimonGrn committed:
Merge pull request #17762 from boubkerbribri/ImportLocalizationTest

Functional tests - Add test 'Import localization pack'

SimonGrn committed:
Merge pull request #17775 from nesrineabdmouleh/functionalCustomerSettingsSendWelcomeEmail

Functional tests - Customer settings enable/disable send email after registration

Commit:
Fix tests identifiers

nesrineabdmouleh committed:
Fix test context

Commit:
Fix sending emails in nightly

soulardromain committed:
Apply suggestions from code review

Co-Authored-By: Progi1984 <progi1984@gmail.com>

soulardromain committed:
Apply suggestions from code review

Co-Authored-By: Progi1984 <progi1984@gmail.com>

rsoulard-prolaser committed:
Apply suggestions from code review

rsoulard-prolaser committed:
Apply suggestions from code review

SimonGrn committed:
Merge pull request #17772 from boubkerbribri/SortAddressesTest

Functional tests - Add test 'Sort Addresses'

SimonGrn committed:
Merge pull request #17824 from boubkerbribri/fixSendEmailForNightly

Functional tests - Fix sending emails in nightly

nesrineabdmouleh committed:
Set display suppliers

nesrineabdmouleh committed:
Add some fixes

PierreRambaud reviewed (dismissed):
LGTM

nesrineabdmouleh committed:
Set display brands

SimonGrn committed:
Merge pull request #17779 from boubkerbribri/AddCategoriesTests

Functional tests - Add new tests 'Sort categories' and 'Change category position'

matks committed:
Merge pull request #17746 from matks/add-ga-instead-of-prettyci

Restore GA for PHP CS Fixer

nesrineabdmouleh committed:
check alert in suppliers page

SimonGrn committed:
Merge pull request #17828 from nesrineabdmouleh/functionalGeneralSetDisplaySuppliers

Functional tests - Shop Parameters> General enable/disable display suppliers

nesrineabdmouleh committed:
Check alert message in brands page

SimonGrn committed:
Merge pull request #17754 from nesrineabdmouleh/functionalProductSettingsDisplayRemainingQuantity

Functional tests - Product settings display remaining quantity in product page

nesrineabdmouleh committed:
Fix some issues

nesrineabdmouleh committed:
Add a little fix

Commit:
fix employee faker

SimonGrn committed:
Merge pull request #17829 from nesrineabdmouleh/functionalGeneralSetDisplayBrands

Functional tests - Shop Parameters> General enable/disable display brands

SimonGrn committed:
Merge pull request #17795 from nesrineabdmouleh/functionalPaymentPreferencesConfigureCurrencyRestrictions

Functional tests - Payment>Preferences Configure currency restriction

Commit:
Improve issue template

Commit:
Improve PR template

matks committed:
Merge pull request #17841 from eternoendless/update-categories

Improve PR template

PierreRambaud committed:
Possible leak information

Make sure the customer is the connected one when changing personnal
data
Make sure addresses are owned by the connected customer when editing
one

Commit:
Remove 1.6.1.x from PR template

SimonGrn committed:
Merge pull request #17836 from boubkerbribri/fixEmployeeFaker

Funtional tests - Fix employee faker

Commit:
Add linkedin

nesrineabdmouleh committed:
Delete check legal free text

matthieu-rolland committed:
round values with computing precision

SimonGrn committed:
Merge pull request #17872 from nesrineabdmouleh/deleteInvoiceCheckLegalFreeText

Functional tests - Invoices > Other options - delete check legal free text from scenario

PierreRambaud committed:
Merge pull request #17785 from davidglezz/patch-11

Fix X-Forwarded-For not supported properly

zuk3975 committed:
use constant for reduction type

PierreRambaud committed:
Merge pull request #17840 from eternoendless/improve-issue-template

Improve issue template

matthieu-rolland committed:
add missing hook in update sql file for 1.7.7.0

SimonGrn committed:
Merge pull request #17699 from nesrineabdmouleh/functionalProductSettingsChooseQuantityBasedOn

Functional Tests - Product settings choose quantity discounts based on

matks committed:
Merge pull request #17884 from zuk3975/fix/todo/catalog-price-rule

Use constant for reduction type in CatalogPriceRuleGridDataFactory

nesrineabdmouleh committed:
Add some fixes

nesrineabdmouleh committed:
Edit future year

arouiadib committed:
Use minimal quantity as default value instead of 1

The GET['quantity_wanted'] doesn't always contain a number. There is a case where it is null, when on initial loading of product page (in FO). Putting 1 as a default value for "Tools::getValue()" is good. But for products with minimal_quantity > 1, this default value should be minimal quantity.

Commit:
Fix nightly 29/02

Commit:
Fix lint errors

jolelievre committed:
Remove useless TinyMCE scripts from brand form pages

SimonGrn committed:
Merge pull request #17910 from boubkerbribri/FixNightly2902

Functional tests - Improving FO tests by adding wait for cart to disappear

atomiix committed:
Merge pull request #17889 from matthieu-rolland/add-hook-update

Add displayAdditionalCustomerAddressFields hook during update

matthieu-rolland committed:
// Changelog 1.7.6.4

Commit:
Click on checkbox GDPR

davidglezz committed:
Remove tabs for summary and description fields in product page

nesrineabdmouleh committed:
Fix the future date creation

SimonGrn committed:
Merge pull request #17960 from boubkerbribri/checkboxGPDR

Functional tests - Accept GDPR if only exist

SimonGrn committed:
Merge pull request #17948 from nesrineabdmouleh/improveGenerateFilesByDate

Functional tests - Fix problem of generate files by date for nightly 29-02

matthieu-rolland committed:
fix getCountQueryBuilder query for orders

nesrineabdmouleh committed:
Refacto test script

nesrineabdmouleh committed:
Add some fixes

mvorisek committed:
Improve array sum

matthieu-rolland committed:
display correct information depending on customer status (guest or not)

arouiadib committed:
Check if combination quantity is superior than combination minimal quantity

IF a combination has a quantity less than the minimum allowed quantity, hide the add-to-cart button

arouiadib committed:
No need now to send a default value equal to minimal quantity if no GET['quantity_wanted']

matthieu-rolland committed:
use is same as twig comparison

PierreRambaud committed:
Merge pull request #16813 from mvorisek/gh_pr_fix_autoincrement

Add support for non-unity MySQL autoincrement

nesrineabdmouleh committed:
Add some fixes

SimonGrn committed:
Merge pull request #17974 from nesrineabdmouleh/refactoDisplayAvailableQuantities

Functional tests - Refacto product settings display available quantities

nesrineabdmouleh committed:
Add some fixes

nesrineabdmouleh committed:
Add some fixes

nesrineabdmouleh committed:
Add some fixes

nesrineabdmouleh committed:
Fix a little issue in the function description

Commit:
Merge branch '1.7.7.x' of github.com:PrestaShop/PrestaShop into merge2develop

zalexki committed:
update password phpdoc

nesrineabdmouleh committed:
Add some fixes

nesrineabdmouleh committed:
Update test context

nesrineabdmouleh committed:
Add some fixes

LouiseBonnard committed:
Remove useless spaces for Ceuta and Melilla

matthieu-rolland committed:
assets build

matks committed:
Remove prettyCI config file

matthieu-rolland committed:
cs fixer

SimonGrn committed:
Merge pull request #17989 from nesrineabdmouleh/refactoDisplayRemainingQuantities

Functional tests - Refacto product settings display remaining quantities

SimonGrn committed:
Merge pull request #17996 from nesrineabdmouleh/refactoCatalogMode

Functional tests - Refacto product settings catalog mode

SimonGrn committed:
Merge pull request #17997 from nesrineabdmouleh/refactoNewDaysNumber

Functional tests- Refacto product settings new days number

SimonGrn committed:
Merge pull request #17998 from nesrineabdmouleh/refactoForceUpdateFriendlyURL

Functional tests - Refacto product settings force update friendly URL

SimonGrn committed:
Merge pull request #18001 from nesrineabdmouleh/refactoDefaultActivationStatus

Functional tests - Refacto product settings default activation status

matthieu-rolland committed:
fix wrong class usage

PierreRambaud committed:
Merge pull request #18022 from LouiseBonnard/develop

Remove useless spaces for Ceuta and Melilla

PierreRambaud committed:
Merge pull request #18017 from zalexki/patch-4

Update Customer-> passwd phpdoc

davidglezz committed:
Fix popover not hidden when changing product tab

davidglezz committed:
Fix Brand dropdown empty value

davidglezz committed:
Update brand placeholder translation domain

davidglezz committed:
Fix layout problem in BO Products category Filter

davidglezz committed:
Change manufacturer by brand

tdavidsonas88 committed:
hotfix

matks committed:
Merge pull request #18012 from PierreRambaud/merge2develop

Merge 1.7.7.x into develop - 06/03/2020

matthieu-rolland committed:
Merge pull request #18019 from matthieu-rolland/merge-into-develop

Merge 1.7.6.x into 1.7.7.x

matthieu-rolland committed:
use proper computing precision

Commit:
Fix menu showing

matthieu-rolland committed:
Merge pull request #18050 from matthieu-rolland/fix-precision-payment-module

Use proper computing precision in payment module

Commit:
Back to default sort in test Sort Products

matthieu-rolland committed:
Merge pull request #18056 from boubkerbribri/FixDefaultSortInProductsPage

Functional tests - Back to default sort in test 'Sort Products'

matks committed:
Fix some license headers for 2020

matks committed:
Merge pull request #18057 from matks/fix-license-headers-again

Fix some license headers for 2020

Commit:
changing name color to hex

davidglezz committed:
use name if legend its empty in img alt attribute

matthieu-rolland committed:
Merge pull request #18029 from matks/remove-pretty-ci

Remove prettyCI config file

tdavidsonas88 committed:
Scenario: Get manufacturer for viewing

tdavidsonas88 committed:
wording changes

jolelievre committed:
Merge pull request #17120 from mvorisek/patch-2

Fix & remove unneeded reference operators

matthieu-rolland committed:
Merge pull request #17973 from matthieu-rolland/fix-order-pagination

Fix pagination on migrated order listing

Progi1984 committed:
Merge pull request #17799 from matthieu-rolland/fix-private-note

Order Page: Fix private note that cannot be emptied

Progi1984 committed:
Merge pull request #17913 from jolelievre/clean-tinymce-scripts

Remove useless TinyMCE scripts from brand form pages

matthieu-rolland committed:
Merge pull request #17804 from matthieu-rolland/fix-order-email-change

Fix bug when changing addresses on virtual product command

Progi1984 committed:
Merge pull request #17438 from NeOMakinG/issue16932-177

Fix submenu not displayed on mobile on classic theme

Progi1984 committed:
Merge pull request #17877 from matthieu-rolland/fix-add-discount

Fix bug when adding a discount on order page

matks committed:
Merge pull request #17901 from arouiadib/fix-17848

The button add to cart shouldn't be enabled when minimal quantity > quantity

matks committed:
Merge pull request #17650 from Progi1984/issue17563

Fixed display for first cell (select all & input) in grid

matks committed:
Merge pull request #17742 from zuk3975/fix/state-toggle

Fix country states input toggle on page load

matks committed:
Merge pull request #18045 from tdavidsonas88/17918-bug-cannot-add-note-for-second-invoice

Fix selector for invoice note manager

Progi1984 committed:
Merge pull request #17985 from matthieu-rolland/fix-guest-order-page

Fix customer informations on order page when customer is a guest

matks committed:
Merge branch '1.7.7.x' of github.com:PrestaShop/PrestaShop into merge2develop-16

matks committed:
Remove committed assets following merge

Arman-Hosseini committed:
Add return to Tools::deleteFile() method

tdavidsonas88 committed:
hardcoded values passed by parameter and specified in the scenario

Progi1984 committed:
Merge pull request #18068 from davidglezz/img-alt

Improve img accessibility

PierreRambaud committed:
Merge pull request #18035 from davidglezz/issue-13513

Fix popover not hidden when changing product tab

Commit:
Merge pull request #18040 from davidglezz/issue-9678

Fix Brand dropdown empty value

PierreRambaud committed:
Merge pull request #18098 from Arman-Hosseini/ps-adblretol

Add return to Tools::deleteFile() method

Progi1984 committed:
Merge pull request #18051 from NeOMakinG/issue17553

Fix Module Catalog page - The Menu "process-icon-dropdown" is not well displayed on mobile by adjusting sass

matks committed:
Add some phpdoc to Tools

matks committed:
Run php-cs-fixer

matks committed:
Fix ESLint

Commit:
Merge pull request #18041 from davidglezz/issue-9551

Fix layout problem in BO Products category Filter

Progi1984 committed:
Merge pull request #18093 from matks/merge2develop-16

Merge 177x into develop - 11/03/2020

PierreRambaud committed:
Merge pull request #18033 from davidglezz/summary-description-untab

Remove tabs for summary and description fields in product page

matks committed:
Fix phpdoc of Tools.php

Co-Authored-By: GoT <PierreRambaud@users.noreply.github.com>

matks committed:
Merge pull request #18077 from tdavidsonas88/behat-manufacturer-for-viewing-handler-test

Add behat test for GetManufacturerForViewingHandler

Progi1984 committed:
Merge pull request #17869 from NeOMakinG/add-linkedin

Add linkedin to classic theme with SVG icon

Progi1984 commented:
Hello @soulardromain , thanks for your contribution, would you mind to rebase your pull request? Thanks

PierreRambaud committed:
Merge pull request #18135 from matks/some-phpdoc

Add some phpdoc to Tools

rsoulard-prolaser committed:
Merge remote-tracking branch 'origin/fix-sort-price' into fix-sort-price

soulardromain commented:
Hello @Progi1984, I think I just mishandled...

Progi1984 commented:
@soulardromain Aouch ! Could you recreate a new PR with your fix ? Thansk :)

soulardromain commented:
Hello @Progi1984, The new PR is #18179

Progi1984 commented:
Ok. I close this one ;)
